docs/book/src/tasks/change-machine-template.md

```diff
@@ -0,0 +1,37 @@
+# Changing Infrastructure Machine Templates
+
+Several different components of Cluster API leverage _infrastructure machine templates_, 
+including `KubeadmControlPlane`, `MachineDeployment`, and `MachineSet`. These 
+`MachineTemplate` resources should be immutable, unless the infrastructure provider 
+documentation indicates otherwise for certain fields (see below for more details).
+
+The correct process for modifying an infrastructure machine template is as follows:
+
+1. Duplicate an existing template.
+    Users can use `kubectl get <MachineTemplateType> <name> -o yaml > file.yaml` 
+    to retrieve a template configuration from a running cluster to serve as a starting 
+    point.
+2. Update the desired fields.
+    Fields that might need to be modified could include the SSH key, the AWS instance 
+    type, or the Azure VM size). Refer to the provider-specific documentation 
+    for more details on the specific fields that each provider requires or accepts.
+3. Give the newly-modified template a new name by modifying the `metadata.name` field 
+    (or by using `metadata.generateName`).
+4. Create the new infrastructure machine template on the API server using `kubectl`.
```

Comment:
If you saved to disk as indicated in step 1, you'll also need to clear out the resource version before trying to send it to the apiserver.

Reply:
@ncdc Should I omit this command entirely? Is there a better way to generate a new machine template?

Reply:
I would just add a parenthetical that says make sure you clear out the RV

Reply:
@ncdc Have a look at this updated wording and let me know what you think (added in 276fe9f).